scheduler/replica_scheduler.go

```diff
@@ -84,25 +83,27 @@ func (rcs *ReplicaScheduler) ScheduleReplica(replica *longhorn.Replica, replicas
 		nodeDisksMap[node.Name] = disks
 	}
 
-	diskCandidates := rcs.getDiskCandidates(nodeCandidates, nodeDisksMap, replicas, volume, true)
+	diskCandidates, verboseErrors := rcs.getDiskCandidates(nodeCandidates, nodeDisksMap, replicas, volume, true)
 
 	// there's no disk that fit for current replica
 	if len(diskCandidates) == 0 {
 		logrus.Errorf("There's no available disk for replica %v, size %v", replica.ObjectMeta.Name, replica.Spec.VolumeSize)
-		return nil, nil
+		return nil, verboseErrors, nil
 	}
 
 	// schedule replica to disk
 	rcs.scheduleReplicaToDisk(replica, diskCandidates)
 
-	return replica, nil
+	return replica, nil, nil
 }
 
-func (rcs *ReplicaScheduler) getNodeCandidates(nodesInfo map[string]*longhorn.Node, schedulingReplica *longhorn.Replica) map[string]*longhorn.Node {
+func (rcs *ReplicaScheduler) getNodeCandidates(nodesInfo map[string]*longhorn.Node, schedulingReplica *longhorn.Replica) (map[string]*longhorn.Node, map[string]struct{}) {
```

Comment:
It seems the returned verbose error type `map[string]struct{}` frequently used here, so I would suggest having a type alias which is better for debugging/development.

Reply:
okay